Source/SwiftLintFramework/Rules/ImplicitlyUnwrappedOptionalRule.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  ImplicitlyUnwrappedOptionalRule.swift
+//  SwiftLint
+//
+//  Created by Siarhei Fedartsou on 17/03/17.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct ImplicitlyUnwrappedOptionalRule: ASTRule, ConfigurationProviderRule, OptInRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "implicitly_unwrapped_optional",
+        name: "Implicitly Unwrapped Optional",
+        description: "Implicitly unwrapped optionals should be avoided when possible.",
+        nonTriggeringExamples: [
+            "@IBOutlet private var label: UILabel!",
+            "@IBOutlet var label: UILabel!",
+            "if !boolean {}",
+            "let int: Int? = 42",
+            "let int: Int? = nil"
+        ],
+        triggeringExamples: [
+            "let label: UILabel!",
+            "let IBOutlet: UILabel!",
+            "let label: IBOutlet!",
+            "let int: Int! = 42",
+            "let int: Int! = nil",
+            "var int: Int! = 42"
+        ]
+    )
+
+    public func validate(file: File, kind: SwiftDeclarationKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard SwiftDeclarationKind.variableKinds().contains(kind) else {
+            return []
+        }
+
+        guard let typeName = dictionary.typeName  else { return [] }
+        guard typeName.hasSuffix("!") else { return [] }
+
+        let isOutlet = dictionary.enclosedSwiftAttributes.contains("source.decl.attribute.iboutlet")
```

Comment:
what do you think about making this configurable? I think by default we should skip outlets, but I know some people who think that IUO shouldn't be used even on outlets.